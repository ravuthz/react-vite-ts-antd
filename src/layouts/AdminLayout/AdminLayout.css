.admin-layout {
  min-height: 100vh;
}

.admin-layout-header {
  padding: 0 16px;
}

.admin-layout-footer {
  padding: 16px;
  text-align: center;
}

.admin-layout-header,
.admin-layout-content,
.admin-layout-footer {
  background: #fff;
}

.admin-layout-content {
  margin: 16px;
  padding: 24px;
  min-height: 280px;
}

.trigger,
.setting {
  font-size: 18px;
  line-height: 64px;
  cursor: pointer;
  transition: color 0.3s;
}

.setting-title {
  font-size: 24px;
  text-decoration: bold;
}

.trigger:hover {
  color: #1890ff;
}

.setting {
  float: right;
}

.logo {
  height: 32px;
  margin: 16px;
  background: rgba(255, 255, 255, 0.3);
  color: #fff;
}

.dark {
  color: #fff;
  background-color: #001529;
}

.fixed-header {
  position: fixed;
  width: 100%;
  z-index: 1;
}

.fixed-sidebar {
  position: fixed;
  height: 100vh;
  overflow: auto;
  left: 0;
  top: 0;
  bottom: 0;
}

.right-sidebar {
  flex: 1 1 auto;
}
