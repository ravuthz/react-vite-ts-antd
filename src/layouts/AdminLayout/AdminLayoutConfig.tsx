import { ReactNode } from "react";

export interface AdminLayoutConfig {
    theme: any;
    showSetting: boolean;
    showFixHeader: boolean;
    showFixSideBar: boolean;
    showSideBarLogo: boolean;
    collapsedSideBar: boolean;
}

export interface AdminLayoutProps {
    children: ReactNode;
    footer?: ReactNode;
    menus?: any[];
}