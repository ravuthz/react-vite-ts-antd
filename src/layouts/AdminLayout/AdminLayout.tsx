import { CheckOutlined, CloseOutlined, MenuFoldOutlined, MenuUnfoldOutlined, SettingOutlined } from '@ant-design/icons';
import { ConfigProvider, Drawer, Layout, List, Menu, Switch } from 'antd';
import { Theme } from 'antd/lib/config-provider/context';
import { atom, useAtom } from 'jotai';
import React, { useCallback } from 'react';
import { AdminLayoutConfig, AdminLayoutProps } from './AdminLayoutConfig';

const { Content, Footer, Header, Sider } = Layout;

import './AdminLayout.css';
// import classes from './AdminLayout.module.css';

const config: AdminLayoutConfig = {
  theme: 'light', // 'dark',
  showSetting: false,
  showFixHeader: false,
  showFixSideBar: false,
  showSideBarLogo: true,
  collapsedSideBar: true
};

const adminLayoutConfigAtom = atom(config);

const SwitchKey: React.FC<any> = (props) => {
  return (
    <Switch
      {...props}
      checkedChildren={<CheckOutlined />}
      unCheckedChildren={<CloseOutlined />}
    />
  );
};

const AdminLayout: React.FC<AdminLayoutProps> = ({ children, footer, menus }) => {
  const [config, setConfig] = useAtom(adminLayoutConfigAtom);

  const { theme, showSetting, showFixHeader, showFixSideBar, showSideBarLogo, collapsedSideBar }: AdminLayoutConfig = config;

  const toggleConfig = useCallback((key: string) => {
    console.log('toggleConfig.key: ', key);
    setConfig((config: any) => ({ ...config, [key]: !config[key] }));
  }, [setConfig]);

  const handleThemeChanage = (checked: boolean) => {
    // setConfig((config: any) => ({ ...config, theme: checked ? 'dark' : 'light' }));

    let color = {};

    // if (checked) {
    //   color = {
    //     primaryColor: '#24292f',
    //     errorColor: '#ff4d4f',
    //     warningColor: '#faad14',
    //     successColor: '#52c41a',
    //     infoColor: '#1890ff'
    //   };
    // } else {
    //   color = {
    //     primaryColor: '#1890ff',
    //     errorColor: '#ff4d4f',
    //     warningColor: '#faad14',
    //     successColor: '#52c41a',
    //     infoColor: '#1890ff'
    //   };
    // }


    ConfigProvider.config({
      theme: {
        ...color
      }
    });

  }

  // className={classes.AdminLayout}
  return (
    <Layout className="admin-layout" data-testid="AdminLayout">
      <Sider theme={config.theme} trigger={null} collapsible collapsed={collapsedSideBar} className={`${showFixSideBar} ? 'fixed-sidebar': ''`}>
        {showSideBarLogo && <div className="logo" />}
        <Menu theme={config.theme} mode="inline" items={menus}></Menu>
      </Sider>
      <Drawer closable={false} placement="right" visible={showSetting} onClose={() => toggleConfig('showSetting')}>
        <List header={<div className="setting-title">Settings</div>} bordered={false}>
          <List.Item key="collapsedSideBar">
            Collapsed SideBar
            <SwitchKey style={{ float: 'right' }} checked={collapsedSideBar} onChange={() => toggleConfig('collapsedSideBar')} />
          </List.Item>
          <List.Item key='showSideBarLogo'>
            Show Icon
            <SwitchKey style={{ float: 'right' }} checked={showSideBarLogo} onChange={() => toggleConfig('showSideBarLogo')} />
          </List.Item>
          <List.Item key="showFixHeader">
            Fixed Header
            <SwitchKey style={{ float: 'right' }} checked={showFixHeader} onChange={() => toggleConfig('showFixHeader')} />
          </List.Item>
          <List.Item key="showFixSideBar">
            Fixed SideBar
            <SwitchKey style={{ float: 'right' }} hecked={showFixSideBar} onChange={() => toggleConfig('showFixSideBar')} />
          </List.Item>
          <List.Item key="theme">
            Theme
            <Switch
              style={{ float: 'right' }}
              checked={theme === 'dark'}
              onChange={handleThemeChanage}
              checkedChildren="Dark"
              unCheckedChildren="Light"
            />
          </List.Item>
        </List>
      </Drawer>
      <Layout className="site-layout">
        <Header className={`admin-layout-header ${theme} ${showFixHeader ? 'fixed-header' : ''} `}>
          {React.createElement(collapsedSideBar ? MenuUnfoldOutlined : MenuFoldOutlined, {
            className: 'trigger',
            onClick: () => toggleConfig('collapsedSideBar'),
          })}
          {React.createElement(SettingOutlined, {
            className: 'setting',
            onClick: () => toggleConfig('showSetting')
          })}
        </Header>
        <Content
          className={`admin-layout-content ${theme}`}
        >
          {children}
        </Content>
        {footer && <Footer className={`admin-layout-footer ${theme}`}>{footer}</Footer>}
      </Layout>
    </Layout>
  );
};

export default AdminLayout;