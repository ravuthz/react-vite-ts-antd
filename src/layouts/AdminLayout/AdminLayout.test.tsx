import { cleanup, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { afterEach, describe, expect, test } from 'vitest';

import AdminLayout from './AdminLayout';

describe('AdminLayout', () => {
    afterEach(cleanup)

    test('Minimal render display expected text', () => {
        render(<AdminLayout children={<h1>AdminLayout without footer</h1>} />);
        expect(screen.getByText('AdminLayout without footer'))
    })

    test('AdminLayout component renders correctly with children props', () => {
        const component = renderer.create(
            <AdminLayout children={<h1>AdminLayout without footer</h1>} />
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('AdminLayout component renders correctly with any props', () => {
        const component = renderer.create(
            <AdminLayout footer="Footer" children={<h1>AdminLayout with footer</h1>} />
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
