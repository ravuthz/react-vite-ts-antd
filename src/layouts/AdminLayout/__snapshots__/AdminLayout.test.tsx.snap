// Vitest Snapshot v1

exports[`AdminLayout > AdminLayout component renders correctly with any props 1`] = `
<section
  className="ant-layout ant-layout-has-sider admin-layout"
  data-testid="AdminLayout"
>
  <aside
    className="ant-layout-sider ant-layout-sider-dark ant-layout-sider-collapsed false ? 'fixed-sidebar': ''"
    style={
      {
        "flex": "0 0 80px",
        "maxWidth": "80px",
        "minWidth": "80px",
        "width": "80px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <div
        className="logo"
      />
      <ul
        className="ant-menu ant-menu-root ant-menu-vertical ant-menu-dark ant-menu-inline-collapsed"
        data-menu-list={true}
        onKeyDown={[Function]}
        role="menu"
        tabIndex={0}
      />
      <div
        aria-hidden={true}
        style={
          {
            "display": "none",
          }
        }
      />
    </div>
  </aside>
  <section
    className="ant-layout site-layout"
  >
    <header
      className="ant-layout-header admin-layout-header dark  "
    >
      <span
        aria-label="menu-unfold"
        className="anticon anticon-menu-unfold trigger"
        onClick={[Function]}
        role="img"
        tabIndex={-1}
      >
        <svg
          aria-hidden="true"
          data-icon="menu-unfold"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 000-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0014.4 7z"
          />
        </svg>
      </span>
      <span
        aria-label="setting"
        className="anticon anticon-setting setting"
        onClick={[Function]}
        role="img"
        tabIndex={-1}
      >
        <svg
          aria-hidden="true"
          data-icon="setting"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
          />
        </svg>
      </span>
    </header>
    <main
      className="ant-layout-content admin-layout-content dark"
    >
      <h1>
        AdminLayout with footer
      </h1>
    </main>
    <footer
      className="ant-layout-footer admin-layout-footer dark"
    >
      Footer
    </footer>
  </section>
</section>
`;

exports[`AdminLayout > AdminLayout component renders correctly with children props 1`] = `
<section
  className="ant-layout admin-layout"
  data-testid="AdminLayout"
>
  <aside
    className="ant-layout-sider ant-layout-sider-dark ant-layout-sider-collapsed false ? 'fixed-sidebar': ''"
    style={
      {
        "flex": "0 0 80px",
        "maxWidth": "80px",
        "minWidth": "80px",
        "width": "80px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <div
        className="logo"
      />
      <ul
        className="ant-menu ant-menu-root ant-menu-vertical ant-menu-dark ant-menu-inline-collapsed"
        data-menu-list={true}
        onKeyDown={[Function]}
        role="menu"
        tabIndex={0}
      />
      <div
        aria-hidden={true}
        style={
          {
            "display": "none",
          }
        }
      />
    </div>
  </aside>
  <section
    className="ant-layout site-layout"
  >
    <header
      className="ant-layout-header admin-layout-header dark  "
    >
      <span
        aria-label="menu-unfold"
        className="anticon anticon-menu-unfold trigger"
        onClick={[Function]}
        role="img"
        tabIndex={-1}
      >
        <svg
          aria-hidden="true"
          data-icon="menu-unfold"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 000-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0014.4 7z"
          />
        </svg>
      </span>
      <span
        aria-label="setting"
        className="anticon anticon-setting setting"
        onClick={[Function]}
        role="img"
        tabIndex={-1}
      >
        <svg
          aria-hidden="true"
          data-icon="setting"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
          />
        </svg>
      </span>
    </header>
    <main
      className="ant-layout-content admin-layout-content dark"
    >
      <h1>
        AdminLayout without footer
      </h1>
    </main>
  </section>
</section>
`;
